src/shims/foreign_items.rs

```diff
@@ -359,6 +441,21 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, EmulateByNameResult<'mir, 'tcx>> {
         let this = self.eval_context_mut();
 
+        let tcx = this.tcx.tcx;
+        // first deal with any external C functions in linked .so files
+        if let Some(local_id) = def_id.as_local() {
+            if let Some(extern_c_fct_rep) = ExternalCFuncDeclRep::from_hir_node(
+                &tcx.hir().get(tcx.hir().local_def_id_to_hir_id(local_id)),
+                link_name,
+            ) {
+                if let Ok(_) =
+                    this.call_and_add_external_c_fct_to_context(extern_c_fct_rep, dest, args)
```

Comment:
The errors should be forwarded with `?` and this entire code here should be gated on there being a `external_c_so_file` specified.